services/ux-backend/handlers/onboardingtokens/handler.go

```diff
@@ -52,3 +59,25 @@ func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
 		klog.Errorf("failed write data to response writer: %v", err)
 	}
 }
+
+func getQuantity(r *http.Request) (*resource.Quantity, error) {
+	var quota = struct {
+		StorageQuota string `json:"storageQuota"`
+	}{}
+	err := json.NewDecoder(r.Body).Decode(&quota)
+	if err != nil && err.Error() != "EOF" {
+		return nil, err
+	}
+
+	// When length is 0 that means request body is empty and quota
+	// is unlimited
+	if len(quota.StorageQuota) == 0 {
+		return nil, nil
+	}
+	quantity, err := resource.ParseQuantity(quota.StorageQuota)
+	if err != nil {
+		return nil, fmt.Errorf("invalid StorageQuota value sent in request body: %v", err)
+	}
+	return &quantity, nil
+
+}
```

Comment:
I am not sure about the logic here. If the UI is passing us a quantity I would expect to see 2 fields in the body. One for the unit and one for the size. But if there is an assumption about the unit then there is no reason to parse it as a quantity, it should be just an int. 

Anything else just add unnecessary complexity to the system

Reply:
UI will be passed as a string in the body with the format `1Gi/1Mi/1Ti` to indicate quantity. I will decode and parse it to validate the correctness of the passed value. Invalid values will look like `1/1gi/1ti`.